harness/determined/core/_train.py

```diff
@@ -68,6 +68,18 @@ def _get_last_validation(self) -> Optional[int]:
         logger.debug(f"_get_last_validation() -> {steps_completed}")
         return steps_completed
 
+    def log_metadata(self, metadata: Dict[str, Any]) -> None:
```

Comment:
AFAICT, on the backend this will merge/update the metadata for the run, right? if so, i think this method should be called `add_metadata`

Reply:
You're absolutely right, `log_metadata` can be a bit misleading from a developer perspective.

The merge logic will be removed prior to merging, so the expected behavior will be that additional calls will overwriting the existing metadata with what's provided in additional calls. I believe @stoksc  had recommended something along the lines of `set_metadata`, does that align with what you'd expect from this function?

In the future, I think if we do decide to have a way to append/update the existing metadata something like `add_metadata` could be used.

Reply:
> additional calls will overwriting the existing metadata with what's provided in additional calls
oh so we're not going to merge at all? so whatever is passed into `set_metadata(metadata={"asdf": 1})` will completely overwrite all existing metadata? if that's the case then yeah i think `set_metadata` is the way to go.

why are we removing the merge logic? it seems better from a user perspective to have add/remove, so that you can call those methods at any point during your Core API code. if we're always overwriting, we're effectively saying you have to keep track of all the metadata you want to add throughout your training code yourself, and only call `set_metadata` once at the end. you wouldn't be able to log it throughout your code, so if your code threw up before the end log, you wouldn't have any metadata. 

not a big deal, but just wondering what's the reason for going with set/overwrite over add/update-merge

Reply:
The motivation to go with set/overwrite had been to provide a more flexible experience for end users and create a smaller set of building blocks that could reproduce the same behavior instead of potentially only supporting niche merge behavior in the event that a user doesn't want the metadata to structured in the way the current merge behavior is configured.

Reply:
spoke with @stoksc a bit on this, `get`/`set` make sense to me now. we shouldn't treat metadata like metrics that should be logged continuously. 

but i think this means we should probably expose a `get` for this, since we're expecting users to manage metadata themselves.